internal/fileutils/fileutils_test.go

```diff
@@ -0,0 +1,156 @@
+package fileutils_test
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/ubuntu/authd/internal/fileutils"
+)
+
+func TestFileExists(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		name            string
+		fileExists      bool
+		parentDirIsFile bool
+
+		wantExists bool
+		wantError  bool
+	}{
+		"Returns_true_when_file_exists":                      {fileExists: true, wantExists: true},
+		"Returns_false_when_file_does_not_exist":             {fileExists: false, wantExists: false},
+		"Returns_false_when_parent_directory_does_not_exist": {fileExists: false, wantExists: false},
+
+		"Error_when_parent_directory_is_a_file": {parentDirIsFile: true, wantError: true},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			tempDir := t.TempDir()
+			path := filepath.Join(tempDir, "file")
+			if tc.fileExists {
+				err := fileutils.Touch(path)
+				require.NoError(t, err, "Touch should not return an error")
+			}
+			if tc.parentDirIsFile {
+				path = filepath.Join(tempDir, "file", "file")
+				err := fileutils.Touch(filepath.Join(tempDir, "file"))
+				require.NoError(t, err, "Touch should not return an error")
+			}
+
+			exists, err := fileutils.FileExists(path)
+			if tc.wantError {
+				require.Error(t, err, "FileExists should return an error")
+			} else {
+				require.NoError(t, err, "FileExists should not return an error")
+			}
+			require.Equal(t, tc.wantExists, exists, "FileExists should return the expected result")
+		})
+	}
+}
+
+func TestIsDirEmpty(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		isEmpty      bool
+		isFile       bool
+		doesNotExist bool
+
+		wantEmpty bool
+		wantError bool
```

Comment:
When it's possible I like to check for specific errors, not sure if that's possible, instead of just error/not-error.

Reply:
IMO we should test what's part of the API contract. The API contract for `fileutils.FileExists` doesn't specify which error type is returned.

Reply:
Mh, I see, nw... Sometimes it's still hard to do that, due to API limitations.